react/features/base/tracks/actions.web.ts

```diff
@@ -317,3 +292,225 @@ export function openAllowToggleCameraDialog(onAllow: Function, initiatorId: stri
+                tracks
+            };
+        })
+        .catch(async error => {
+            const errors = {} as IInitialTracksErrors;
+
+            if (error.name === JitsiTrackErrors.TIMEOUT && !browser.isElectron()) {
+                errors.audioAndVideoError = error;
+
+                return {
+                    errors,
+                    tracks: []
+                };
+            }
+
+            // Retry with separate gUM calls.
+            const gUMPromises = [];
+            const tracks: any[] | PromiseLike<any[]> = [];
+
+            if (devices.includes(MEDIA_TYPE.AUDIO)) {
+                gUMPromises.push(createLocalTracksF(audioOptions));
+            }
+
+            if (devices.includes(MEDIA_TYPE.VIDEO)) {
+                gUMPromises.push(createLocalTracksF({
+                    devices: [ MEDIA_TYPE.VIDEO ],
+                    timeout,
+                    firePermissionPromptIsShownEvent
+                }));
+            }
+
+            const results = await Promise.allSettled(gUMPromises);
+            let errorMsg;
+
+            results.forEach((result, idx) => {
+                if (result.status === 'fulfilled') {
+                    tracks.push(result.value[0]);
+                } else {
+                    errorMsg = result.reason;
+                    const isAudio = idx === 0;
+
+                    logger.error(`${isAudio ? 'Audio' : 'Video'} track creation failed with error ${errorMsg}`);
+                    if (isAudio) {
+                        errors.audioOnlyError = errorMsg;
+                    } else {
+                        errors.videoOnlyError = errorMsg;
+                    }
+                }
+            });
+
+            if (errors.audioOnlyError && errors.videoOnlyError) {
+                errors.audioAndVideoError = errorMsg;
+            }
+
+            return {
+                tracks,
+                errors
+            };
+        });
+    };
+}
+
+/**
+ * Displays error notifications according to the state carried by the passed {@code errors} object.
+ *
+ * @param {InitialTracksErrors} errors - The errors (if any).
+ * @returns {Function}
+ * @private
+ */
+export function displayErrorsForCreateInitialLocalTracks(errors: IInitialTracksErrors) {
+    return (dispatch: IStore['dispatch']) => {
+        const {
+            audioAndVideoError,
+            audioOnlyError,
+            screenSharingError,
+            videoOnlyError
+        } = errors;
+
+        // FIXME If there will be microphone error it will cover any screensharing dialog, but it's still better than in
+        // the reverse order where the screensharing dialog will sometimes be closing the microphone alert
+        // ($.prompt.close(); is called). Need to figure out dialogs chaining to fix that.
+        if (screenSharingError) {
+            dispatch(handleScreenSharingError(screenSharingError, NOTIFICATION_TIMEOUT_TYPE.LONG));
+        }
+        if (audioAndVideoError || audioOnlyError) {
```

Comment:
Good point! This code is moved from conference.js but generally it assumes that we request audio+video always which obviously is not the case. Will fix it!

Reply:
Simplified the whole logic. For the displaying error notification we don't need `audioAndVideoError` at all because in this case we will always have the `audioOnlyError` and `videoOnlyError` as well. Except for the timeout case which I improved a bit and we will also have `audioOnlyError` and/or `videoOnlyError`. Eventually we can even get rid of the `audioAndVideoError` but I see it is used for the prejoin and don't want to polute the PR.